emcache/_cython/protocol/ascii/multi_line_parser.pyx

```diff
@@ -26,20 +26,21 @@ cdef class AsciiMultiLineParser:
 
     def feed_data(self, bytes buffer_):
         cdef int len_
-        cdef char* c_buffer
 
         self.buffer_.extend(buffer_)
 
         len_ = len(self.buffer_)
         if len_ < 1:
             return
         
-        c_buffer = self.buffer_
+        cdef char* c_buffer = self.buffer_
         if len_ >= 5:
             if strcmp(<const char *> c_buffer + (len_ - 5), END) == 0:
                 self._parse(len_)
                 self.future.set_result(None)
-                return
+            elif self.buffer_.startswith(b"CLIENT_ERROR"):
```

Comment:
If possible can we do the same as the previous line? something like

```python
cdef const char* CLIENT_ERROR = "CLIENT_ERROR\r\n"
.....
if len_ >= 14:
  if strcmp(<const char *> c_buffer + (len_ - 15), CLIENT_ERROR) == 0:
    self.future.set_exception(CommandError("...."))
```

Reply:
> @pfreixes Your option is not suitable. We return an error here and not [handling](https://github.com/emcache/emcache/pull/146/files#diff-07f845da2abf106971aad8cdbb60c413d2cc88fe54d91bb4cebb28b3e4c43de4R173)

Im not getting why not, if we just put a `set_exception` the future should be woken up with an exception, thats all.

What am I missing?

Reply:
You're assuming you'll get an error [here](https://github.com/emcache/emcache/pull/146/files/4a09e7d2fe39bef962400be84dd783959e2aea82#diff-d5918e537eadb7095bd9b366b7a19681b87d8d0f87683d5d4007a6fa4d5e861cR329)
However, I did exactly what you wanted and received a TimeoutError because I get the error here and do not process it [here](https://github.com/emcache/emcache/pull/146/files#diff-07f845da2abf106971aad8cdbb60c413d2cc88fe54d91bb4cebb28b3e4c43de4R173)

Reply:
Why doesn't the original option suit you?

Reply:
OR.... stop.... please wait

Reply:
Do you want to add CommandError handling here?
```python
    async def _extract_multi_line_data(self, data: bytes):
        try:
            future = self._loop.create_future()
            parser = cyemcache.AsciiMultiLineParser(future)
            self._parser = parser
            self._transport.write(data)
            await future
            keys, values, flags, cas = parser.keys(), parser.values(), parser.flags(), parser.cas()
            return keys, values, flags, cas
        finally:
            self._parser = None
```

Reply:
Do you want to transfer all errors to Сython? After all, we raise errors at the level of client command calls

Reply:
what do I mean? you create errors at the client level and not in cython. Your thoughts?